lib/search/worker/index.ts

```diff
@@ -0,0 +1,127 @@
+import { getTerms } from '../../utils/filter-notes';
+
+import * as T from '../../types';
+
+const notes: Map<
+  T.EntityId,
+  [T.EntityId, T.Note & { tags: Set<T.TagName> }]
+> = new Map();
+
+self.onmessage = bootEvent => {
+  const mainApp: MessagePort | undefined = bootEvent.ports[0];
+
+  if (!mainApp) {
+    // bail if we don't get a custom port
+    return;
+  }
+
+  let searchQuery = '';
+  let searchTerms: string[] = [];
+  let filterTags = new Set<T.TagName>();
+  let showTrash = false;
+
+  const tagsFromSearch = (query: string) => {
+    const tagPattern = /(?:\btag:)([^\s,]+)/g;
+    const searchTags = new Set<T.TagName>();
+    let match;
+    while ((match = tagPattern.exec(query)) !== null) {
+      searchTags.add(match[1].toLocaleLowerCase());
+    }
+    return searchTags;
+  };
+
+  const updateFilter = (scope = 'quickSearch') => {
+    const tic = performance.now();
```

Comment:
"tick" probably?

Reply:
Common practice I picked up in my MATLAB days: tic and toc - l like their symmetry

Reply:
It's just that "tic" has another meaning and it's not a good one..

Reply:
I don't really feel strongly about it though